packages/@aws-cdk/aws-iotevents/test/detector-model.test.ts

```diff
@@ -95,6 +95,65 @@ test('can set physical name', () => {
   });
 });
 
+test('can set description', () => {
+  // WHEN
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    detectorModelDescription: 'test-detector-model-description',
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnter: [{
+        eventName: 'test-eventName',
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    DetectorModelDescription: 'test-detector-model-description',
+  });
+});
+
+test('can set evaluationMethod', () => {
+  // WHEN
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    detectorModelDescription: 'test-detector-model-description',
+    evaluationMethod: iotevents.EvaluationMethod.SERIAL,
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnter: [{
+        eventName: 'test-eventName',
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    EvaluationMethod: 'SERIAL',
+  });
+});
+
+test('can set key', () => {
+  // WHEN
+  new iotevents.DetectorModel(stack, 'MyDetectorModel', {
+    detectorModelDescription: 'test-detector-model-description',
+    key: 'payload.deviceId',
+    initialState: new iotevents.State({
+      stateName: 'test-state',
+      onEnter: [{
+        eventName: 'test-eventName',
+        condition: iotevents.Expression.fromString('test-eventCondition'),
+      }],
+    }),
+  });
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IoTEvents::DetectorModel', {
+    Key: 'payload.deviceId',
+  });
+});
```

Comment:
Let's just do all 3 in one test. Spreading them out like that just makes the tests longer, it doesn't really improve coverage in any way.